terminus-ui/cohort-date-range/src/cohort-date-range.component.spec.ts

```diff
@@ -0,0 +1,139 @@
+import {
+  Component,
+  Provider,
+  Type, ViewChild,
+} from '@angular/core';
+import {
+  ComponentFixture, fakeAsync, tick,
+} from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import {
+  createComponent as createComponentInner, dispatchKeyboardEvent,
+  typeInElement,
+} from '@terminus/ngx-tools/testing';
+import * as testComponents from '@terminus/ui/cohort-date-range/testing';
+// eslint-disable-next-line no-duplicate-imports
+import {
+  getAllDateRangeDebugElements, getRangeInputInstances,
+} from '@terminus/ui/date-range/testing';
+
+import { By } from '@angular/platform-browser';
+import { KEYS } from '@terminus/ngx-tools';
+import {
+  TsDateRangeComponent,
+  TsDateRangeModule,
+} from '@terminus/ui/date-range';
+import {
+  createKeydownEvent, getSelectElement, getSelectInstance,
+  getSelectTriggerElement,
+} from '@terminus/ui/select/testing';
+import {
+  TsCohortDateRangeComponent, TsCohortDateRangeModule,
+} from './cohort-date-range.module';
+
+
+describe(`TsCohortDateRangeComponent`, () => {
+  let fixture;
+  let hostInstance;
+  let startInputInstance;
+  let formFieldElement;
+  let testComponent: testComponents.Basic;
+  let cohortDebugElement;
+  let cohortInstance;
+
+  function setupComponent(exampleComponent) {
+    fixture = createComponent(testComponents[exampleComponent]);
+    fixture.detectChanges();
+    hostInstance = fixture.componentInstance;
+    testComponent = fixture.debugElement.componentInstance;
+    fixture.detectChanges();
+    startInputInstance = getRangeInputInstances(fixture)[0];
+    formFieldElement = fixture.debugElement.query(By.css('.ts-form-field'));
+    cohortDebugElement = fixture.debugElement.query(By.directive(TsCohortDateRangeComponent));
+    cohortInstance = cohortDebugElement.componentInstance;
+  }
```

Comment:
Any specific reason we've started defining all these items up front vs creating helper functions? I've found it much slower to work with personally. But open to discussing it if you're feeling the opposite

Reply:
Personally I like them inside one file, when they're relatively small. It's just much faster for me to go back and forth referencing them. Putting them in another file seems to bring some overhead. But I'm not against putting it in a file if that's the convention we're going forward with. Currently the ones in a separate file are bigger/more complicated components like input, select, autocomplete etc...

Reply:
I'm not entirely sure I follow.. I wasn't saying we should split up files. Just wondering if there was a specific drive that has us using standard queries and saving references to all the things at the top vs using helper functions like we started with select/autocomplete/etc.

Regardless of where they live I do believe using helper functions is a huge productivity boost for us and the consuming teams.

Reply:
I misunderstood.. you are referring them to exported helper function versus inline definition. yeah I definitely can convert those.